src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -745,3 +745,29 @@ TEST_F(ControllerScriptEngineLegacyTest, screenWillSentRawDataIfConfigured) {
     ASSERT_ALL_EXPECTED_MSG();
 }
 #endif
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetUndefinedOnUnknownCharset) {
+    ASSERT_TRUE(evaluate("engine.convertCharset('NULL', 'Hello!')").isUndefined());
+}
+
+TEST_F(ControllerScriptEngineLegacyTest, convertCharsetCorrectValueWellKnown) {
+    const auto result = evaluate(R"Javascript(
+        Array.from(
+            new Uint8Array(
+                engine.convertCharset(engine.WellKnownCharsets.LATIN_9, "Hello!")))
+        .map(it => it.toString(16))
+        .join(",")
+    )Javascript");
+    ASSERT_QSTRING_EQ(result.toString(), "48,65,6c,6c,6f,21");
+}
```

Comment:
I'd prefer something like this:
```suggestion
    const auto result = evaluate(R"Javascript(
        engine.convertCharset(engine.WellKnownCharsets.LATIN_9, "Hello!"))
    )Javascript");
    EXPECT_EQ(jsEngine()->toScriptValue<QByteArray>(result), QByteArrayLiteral("Hello!"));
}
```

Reply:
Wait, how can that work? There shouldn't be the same bytes in the two arrays 🤔

Reply:
well in this case there should be because because were using only characters which are the same in ASCII and LATIN_9, right? For others you could manually create a string with the correct bytes `\x40\x41`. See https://en.cppreference.com/w/cpp/language/escape

Reply:
I'm not sure if that works, but it be nice if you could experiment with it a little.